pkg/network/node/vxlan_monitor.go

```diff
@@ -256,6 +261,18 @@ func (evm *egressVXLANMonitor) check(retryOnly bool) bool {
 	return retry
 }
 
+func (evm *egressVXLANMonitor) updateNode(node *egressVXLANNode) {
+	evm.updateNodesMux.Lock()
+	defer evm.updateNodesMux.Unlock()
+	evm.updateNodes[node.nodeIP] = node
+
+	// Don't update the updates channel if it already contains an entry.
+	// Otherwise we'll block this goroutine.
+	if len(evm.updates) == 0 {
+		evm.updates <- struct{}{}
```

Comment:
The better idiom is

    select {
            case evm.updates <- struct{}{}:
            default:
    }

which is race-proof. (Which, admittedly, doesn't actually matter here because you have a mutex too, but...)